testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/health/MicroProfileHealthApplicationReadyTestBase.java

```diff
@@ -100,4 +112,57 @@ public void testHealthCheckAfterUndeployment() throws Exception {
         checkGlobalOutcome(managementClient, "check-ready", false, null);
     }
 
+    static void testHttpEndPoint(String healthURL, boolean mustBeUP, String probeName) throws IOException {
+
+       try (CloseableHttpClient client = HttpClients.createDefault()) {
+
+            CloseableHttpResponse resp = client.execute(new HttpGet(healthURL));
+            String content = EntityUtils.toString(resp.getEntity());
+            resp.close();
+
+            assertEquals(content, mustBeUP ? 200 : 503, resp.getStatusLine().getStatusCode());
+
+
+            try (
+                    JsonReader jsonReader = Json.createReader(new StringReader(content))
+            ) {
+                JsonObject payload = jsonReader.readObject();
+                String outcome = payload.getString("status");
+                assertEquals(mustBeUP ? "UP": "DOWN", outcome);
+
+                if (probeName != null) {
+                    for (JsonValue check : payload.getJsonArray("checks")) {
+                        if (probeName.equals(check.asJsonObject().getString("name"))) {
+                            // probe name found
+                            assertEquals(mustBeUP ? "UP" : "DOWN", check.asJsonObject().getString("status"));
+                            return;
+                        }
+                    }
+                    fail("Probe named " + probeName + " not found in " + content);
+                }
+            }
+        }
+    }
+
+    static void testHttpOperation(ModelNode response, boolean mustBeUP, String probeName){
```

Comment:
This is a management operation, not HTTP

Reply:
Changes done. Thanks for letting me know.